Assets/MixedRealityToolkit.Extensions/Socketer/Scripts/IncomingMessage.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.Socketer
+{
+    /// <summary>
+    /// Helper class that wraps an incoming message
+    /// </summary>
+    public class IncomingMessage
+    {
+        public IncomingMessage(SocketEndpoint endpoint, byte[] data, int size)
```

Comment:
should this take a MessageEvent or am i missing something?

Reply:
The socket can only receive bytes, and wrapping the data in a message event could be expensive due to frivolous copying

Reply:
The worth in changing this doesn't necessarily seem worth the time required for testing. I won't address this for now.